internal/controller/databasecluster_controller.go

```diff
@@ -369,8 +369,24 @@ func (r *DatabaseClusterReconciler) reconcileLabels(
 
 	updated[databaseClusterNameLabel] = database.Name
 	if database.Spec.DataSource != nil {
-		updated[fmt.Sprintf(backupStorageNameLabelTmpl, database.Spec.DataSource.DBClusterBackupName)] = backupStorageLabelValue
+		// need to obtain backupStorageName by .spec.dataSource.dbClusterBackupName.dbClusterBackupName
+		dbBackup := &everestv1alpha1.DatabaseClusterBackup{}
+		err := r.Get(ctx, types.NamespacedName{
+			Name:      database.Spec.DataSource.DBClusterBackupName,
+			Namespace: database.Namespace,
+		}, dbBackup)
+		if err != nil {
+			return errors.Join(err, fmt.Errorf("could not get DB backup '%s' to obtain backup storage name", database.Spec.DataSource.DBClusterBackupName))
+		}
```

Comment:
the `database.Spec.DataSource.DBClusterBackupName` can be empty (see [here](https://github.com/percona/everest-operator/blob/a3ec50251812e907a7451573fc8c81c72d1ea81a/internal/controller/common/helper.go#L316-L319)), we need to handle it so no error would be returned in that case.

Reply:
done

Reply:
Now the `database.Spec.DataSource.BackupSource.BackupStorageName` (lines 385-387) is handled only if `database.Spec.DataSource.DBClusterBackupName != ""` however the values are independent. 

I know the relations are pretty confusing, let me describe it in more details: 
The `database.Spec.DataSource` is used when a new DB wants to populate itself with some data during creation. So the field could be empty if we want an empty DB. However if the field is not empty it could contain either:  
- `DBClusterBackupName` - the name of the backup CR with the data we want 
- `BackupSource` - the information about the storage (name, path) that contains the actual backup files.